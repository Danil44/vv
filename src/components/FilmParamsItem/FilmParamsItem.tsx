import React from 'react';
import styled from 'styled-components';
import { css as styledCss } from 'styled-components';
import { lightGrey } from '../../colors';
import { jsx, css } from '@emotion/core';
/**@jsx jsx */

const baseSpan = styledCss`
  font-size: 1em;
  font-weight: 400;
  display:block;
`;

const OptionName = styled.span`
  ${baseSpan};
  color: ${lightGrey};
  text-transform: uppercase;
  width: 100%;
  margin-bottom: 10px;
`;

const OptionValue = styled.span`
  ${baseSpan}
  color:#fff;
`;

type FilmParamsItemProps = {
  name: string;
  value: string | number;
};

export default function FilmParamsItem({ name, value }: FilmParamsItemProps) {
  return (
    <li
      css={css`
        width: calc(33.33% - 5px);
        border-top: 2px solid;
        padding-top: 10px;
      `}
    >
      <OptionName data-testid="option-name">{name}</OptionName>
      <OptionValue data-testid="option-value">{value}</OptionValue>
    </li>
  );
}
