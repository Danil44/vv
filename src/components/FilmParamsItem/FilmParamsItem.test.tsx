import React from 'react';
import { ReactWrapper, mount } from 'enzyme';
import FilmParamsItem from './FilmParamsItem';
import { render } from '@testing-library/react';

describe('<FilmParamsItem />', () => {
  let wrapper: ReactWrapper;

  beforeEach(() => {
    wrapper = mount(<FilmParamsItem name="" value="" />);
  });

  it('has not changed', () => expect(wrapper.html()).toMatchSnapshot());

  it('should render the params name and value', () => {
    const { getByTestId } = render(
      <FilmParamsItem name="Length" value="1hour 50minutes" />,
    );

    expect(getByTestId('option-name').textContent).toBe('Length');
    expect(getByTestId('option-value').textContent).toBe('1hour 50minutes');
  });
});
