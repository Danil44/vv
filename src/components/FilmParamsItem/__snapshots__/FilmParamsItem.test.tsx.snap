// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmParamsItem /> has not changed 1`] = `"<li class=\\"css-j5wl32-FilmParamsItem\\"><span data-testid=\\"option-name\\" class=\\"sc-AxjAm koZgQB\\"></span><span data-testid=\\"option-value\\" class=\\"sc-AxirZ kGLtmA\\"></span></li>"`;
